ext/spl/spl_array.stub.php

```diff
@@ -246,3 +248,45 @@ public function hasChildren() {}
     /** @return RecursiveArrayIterator|null */
     public function getChildren() {}
 }
+
+} // namespace
+
+namespace Spl {
+
+final class ForwardArrayIterator
+    implements \Countable, \Iterator
+{
+    public function __construct(array $array) {}
+
+    public function count(): int {}
+    public function rewind(): void {}
+    public function valid(): bool {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    public function next(): void {}
+}
+
+final class ReverseArrayIterator
+    implements \Countable, \Iterator
+{
+    public function __construct(array $array) {}
+
+    public function count(): int {}
+    public function rewind(): void {}
+    public function valid(): bool {}
+
+    /** @return mixed */
+    public function key() {}
```

Comment:
nit: My preference for static analysis/IDEs/opcache would be to set the return type to `int|string`, which can be updated if arrays get changed to other key types such as object.